backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/impexp/ModularContractLogImportProcess.java

```diff
@@ -129,106 +93,28 @@ protected ImportRecordResult importRecord(
-		if (logEntryCreateRequest.isPresent())
-		{
-			modularContractLogDAO.create(logEntryCreateRequest.get());
-			return ImportRecordResult.Inserted;
-		}
-
-		return ImportRecordResult.Nothing;
+		createImportModCntrLog(importRecord);
+		return ImportRecordResult.Inserted;
 	}
 
-	private Optional<LogEntryCreateRequest> getLogEntryCreateRequestFrom(@NonNull final I_I_ModCntr_Log record)
+	private boolean isReadyForImport(@NonNull final I_I_ModCntr_Log importRecord)
 	{
-		final YearId harvestingYearId = YearId.ofRepoIdOrNull(record.getHarvesting_Year_ID());
-		final Optional<ModuleConfig> config = getModuleConfigFrom(record);
+		final YearId harvestingYearId = YearId.ofRepoIdOrNull(importRecord.getHarvesting_Year_ID());
 
-		if (Objects.isNull(harvestingYearId) || config.isEmpty())
-		{
-			throw new AdempiereException("No Contract Config (Contract Settings / Contract Module) Match");
-		}
-
-		final ProductPrice productPrice = getPriceActual(record);
-		return Optional.of(LogEntryCreateRequest.builder()
-								   //
-								   .referencedRecord(TableRecordReference.of(importModCntrLogTableID, record.getI_ModCntr_Log_ID()))
-								   .configId(config.get().getId())
-								   .year(harvestingYearId)
-								   .transactionDate(LocalDateAndOrgId.ofTimestamp(record.getDateTrx(), OrgId.ofRepoId(record.getAD_Org_ID()), orgDAO::getTimeZone))
-								   .soTrx(SOTrx.ofBoolean(record.isSOTrx()))
-								   .documentType(LogEntryDocumentType.IMPORT_LOG) // dev note : all imported logs should have ImportLog as doctype
-								   .contractType(LogEntryContractType.MODULAR_CONTRACT)
-								   //
-								   .contractId(FlatrateTermId.ofRepoIdOrNull(record.getC_Flatrate_Term_ID()))
-								   .collectionPointBPartnerId(BPartnerId.ofRepoIdOrNull(record.getCollectionPoint_BPartner_ID()))
-								   .producerBPartnerId(BPartnerId.ofRepoIdOrNull(record.getProducer_BPartner_ID()))
-								   .invoicingBPartnerId(BPartnerId.ofRepoIdOrNull(record.getBill_BPartner_ID()))
-								   .invoicingGroupId(InvoicingGroupId.ofRepoIdOrNull(record.getModCntr_InvoicingGroup_ID()))
-								   .productId(ProductId.ofRepoIdOrNull(record.getM_Product_ID()))
-								   .quantity(Quantity.ofNullable(record.getQty(), uomDAO.getById(record.getC_UOM_ID())))
-								   .priceActual(productPrice)
-								   .amount(Money.ofOrNull(record.getAmount(), CurrencyId.ofRepoIdOrNull(record.getC_Currency_ID())))
-								   .warehouseId(WarehouseId.ofRepoIdOrNull(record.getM_Warehouse_ID()))
-								   .processed(false)
-								   .description(record.getDescription())
-								   //
-								   .build());
-	}
-
-	private boolean isImportLog(@NonNull final I_I_ModCntr_Log importRecord)
-	{
-		return Objects.equals(importRecord.getModCntr_Log_DocumentType(), X_I_ModCntr_Log.MODCNTR_LOG_DOCUMENTTYPE_ImportLog)
+		return !Objects.isNull(harvestingYearId) && Objects.equals(importRecord.getModCntr_Log_DocumentType(), X_I_ModCntr_Log.MODCNTR_LOG_DOCUMENTTYPE_ImportLog)
```

Comment:
why not simply use `harvestingYearId != null`

Reply:
done.